Src/AutoFixture/MailAddressGenerator.cs

```diff
@@ -31,16 +31,43 @@ public class MailAddressGenerator : ISpecimenBuilder
         /// </remarks>
         public object Create(object request, ISpecimenContext context)
         {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
             if (!typeof(MailAddress).Equals(request))
             {
                 return new NoSpecimen(request);
             }
 
-            var name = Guid.NewGuid().ToString();
-            var host = fictitiousDomains[(uint)name.GetHashCode() % 3];
+            try
+            {
+                return TryCreateMailAddress(request, context);
+            }           
+            catch (ArgumentException)

```

Comment:
Which error case does this `catch` block handle? When would `TryCreateMailAddress` throw an `ArgumentException`?